### ISSUE TITLE
Bug: Should not consume several items at once only so I added them at once

### ISSUE THREAD

**Issue by ybbus** [grocy/grocy#2388]
Hi.

I just found out that when I buy multiple items of the same type (e.g. 3 x Ketchup) I scan just one of these items and provide the quantity (3) for adding all of them to the stock.

The problem now is, that I of course do not want to consume all of them at once, when scanning such an item, but the interface proposes automatically to consume all three items when I scan one Ketchup.

This is very misleading and often leads to accidentally consuming all of the items at once.

This is how it looks in stock when I scan one item and provide 2 as quantity.
(The other item was scanned individually):

![image](https://github.com/grocy/grocy/assets/1547663/79d7914b-15d4-479d-a1ac-98ae0abf4817)

![image](https://github.com/grocy/grocy/assets/1547663/01c52a4e-a20e-4d15-98aa-92703dead86d)

When I scan each item individually it works, but hopefully I don't have to do this (I often by multiple items of the same kind).

Is there any way to change this behavior?

Btw: The "quick consumption amount" is set correctly for the master data of the product:

![image](https://github.com/grocy/grocy/assets/1547663/ffb483a9-8e28-44b5-9522-47e119103d19)

The same is btw. true for buying. When I again buy the product, the amount is automatically set to the last amount.
This also often leads to storing the wrong amount in the stock. E.g. If I filled up the stock with 3 items, next time I only rebuy one item of that kind. But it automatically sets the quantity again to 3. For me this seems to be too smart and should always propose only one new item to be added, since this can quickly be overseen and the stock gets out of sync

Thanks for your help.

**Comment by berrnd** [comment]
> Is there any way to change this behavior?

Yes. When using stock entry labels per unit:

![grafik](https://github.com/grocy/grocy/assets/2161815/0ca93d5c-711a-4112-be98-be0e04937ea7)

&nbsp;

All that sounds for me that you mix up that barcodes on a product are not unique - on every bottle of ketchup is the same barcode, so this can never include an amount and therefore can't refer to a concrete stock entry in any way. Grocy automatically combines equal stock entries, the stock entry label (Grocycode) is a property part of what makes a stock entry unique.

&nbsp;

> But it automatically sets the quantity again to 3. For me this seems to be too smart

There is no magic behind this, what is prefilled after scanning a barcode is just the amount provided with it. Leave that empty or set it to 1 and voila:

![grafik](https://github.com/grocy/grocy/assets/2161815/648ca4d1-29a5-4023-9aec-504b522c9e66)

&nbsp;

> I scan just one of these items and provide the quantity (3) for adding all of them to the stock.

Clarifying again: This adds one stock entry with the amount of 3. If you want uniqueness across them, you need to define a unique property across them, e.g. by using a stock entry label per unit. A barcode always only references a product, never a stock entry. A (stock entry) Grocycode (= stock entry label) can do that.

&nbsp;

In conclusion I don't see a bug on this anywhere. The appropriate place for general usage questions is the [r/grocy subreddit](https://www.reddit.com/r/grocy) as [mentioned in README](https://github.com/grocy/grocy#questions--help--bug-reports--feature-requests) or see maybe [here](https://www.reddit.com/r/grocy/comments/17t9i2x/comment/k8wldh5/?context=3) when there is more confusion about general Grocy stock management basics.

**Comment by berrnd** [comment]
Additional hint:

Never set an amount on a barcode and set these two [stock settings](https://demo-prerelease.grocy.info/stocksettings) to `1` to never get anything else prefilled than that on [/purchase](https://demo-prerelease.grocy.info/purchase) and [/consume](https://demo-prerelease.grocy.info/consume).

![grafik](https://github.com/grocy/grocy/assets/2161815/49f3b1cc-e798-49e8-bb00-ea901f5d2d30)

What you want is essentially, when I get it right, that every single stock entry has an amount of `1` - of course you can organize Grocy like that by what I just pointed out, but generally this won't be changed and is by far not a bug. This would practically remove the amount field anywhere, since everything everywhere is `1` then.
